### DISCUSSION TITLE
MegaMenu bug with more than 4 sub-items

### DISCUSSION THREAD

**Discussion by lorenzocas** [primefaces/community#2705] [PrimeNG]
Only the first subitem is rendered (should be 5). if I remove the last sub items all the sub items are rendered correctly.

Please check the sample on stackblitz.
https://stackblitz.com/edit/5wmdfd?file=src%2Fapp%2Fmega-menu-basic-demo.ts

Thanks
Best Regards

**Comment by qwadrox** [comment]
@lorenzocas Hi. I don't fully understand your question. I checked the example and it looks fine. It renders all items: 
<img width="415" alt="image" src="https://github.com/user-attachments/assets/eee15d11-0e43-40f7-baed-9c56c61709ef">

MegaMenu displays submenus together. Are you looking forward to something like this, with nested structure:
 
<img width="804" alt="image" src="https://github.com/user-attachments/assets/6ad8c5ed-0eca-4575-8a1d-134384c76cbc">

If yes, use MenuBar instead.

**Comment by lorenzocas** [comment]
Yes, Thanks.
I already fixed the sample thanks to the help of a discord user.

Best regards
